drivers/gpio/gpio_intel_apl.c

```diff
@@ -243,7 +243,7 @@ static int gpio_intel_apl_config(struct device *dev, int access_op,
 	/* setup interrupt if desired */
 	if (flags & GPIO_INT) {
 		/* invert signal for interrupt controller */
-		if (flags & GPIO_INT_ACTIVE_LOW) {
+		if (!(flags & GPIO_INT_ACTIVE_HIGH)) {
```

Comment:
Can we make this a bit more explicit:

`((flags & GPIO_INT_ACTIVE_HIGH)) == GPIO_INT_ACTIVE_LOW) {`

Reply:
If we do it that way, the "proper" way to do it would be to have a GPIO_INT_ACTIVE_MASK macro to select the field, which then gets tested against _LOW and _HIGH values.

But it's a 1-bit boolean, so I think having a single name for the bit is fine.  If anything I think the bug in our API is that GPIO_INT_ACTIVE_LOW exists at all.  But API cleanup is out of scope now I'm sure.

Reply:
agreed, post 1.14, I'm thinking we should have a set of macro's since I've seen someone coverity issues like this one.

Reply:
Pushed a new update for this.

I remember there was a PR about creating a trigger mask. I can't seem to find it now.

Reply:
> I remember there was a PR about creating a trigger mask. I can't seem to find it now.

I think in the future we should have helper functions like:

`is_gpio_active_high(x)`
`is_gpio_active_low(x)`

and remove the guess work from the drivers.